packages/amplify-appsync-simulator/src/resolvers/unit-resolver.ts

```diff
@@ -22,6 +23,11 @@ export class AppSyncUnitResolver extends AppSyncBaseResolver {
     const requestMappingTemplate = this.getRequestMappingTemplate();
     const responseMappingTemplate = this.getResponseMappingTemplate();
     const dataLoader = this.simulatorContext.getDataLoader(this.config.dataSourceName);
+
+    // Populate args.query for HTTP data sources.
+    const query = this.getSelectionSet(info.operation?.selectionSet?.selections, args.fieldName);
```

Comment:
Hey Colin! The reason I have this here instead of the data loader class is because the generated request resolver expects the query in `$ctx.args.query`:
```vtl
// Query.listPosts.req.vtl
#set( $headers = $utils.http.copyHeaders($ctx.request.headers) )
$util.qr($headers.put("accept-encoding", "application/json"))
$util.qr($headers.put("SomeHeader", "SomeValue"))
{
  "version": "2018-05-29",
  "method": "GET",
  "resourcePath": "/",
  "params": {
      "query": $util.toJson($ctx.args.query),
      "headers": $util.toJson($headers)
  }
}
```
If I move this logic to the data loader `load` function, it would be fine for this generated resolver, but wouldn't it be a problem if someone made a custom resolver that manipulates `$ctx.args.query`?